lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -541,61 +553,77 @@ sub variant_exists {
         next;
     }
 
-    # Precise match for snv or small indels - end does not make a difference 
-    if ($sv == 0 && ($seq_region_start != $new_start) && ($seq_region_end != $new_end)) {
+    # Precise match for snv or small indels - end does not make a difference
+    if ($sv == 0 && ($seq_region_start != $start_pos) && ($seq_region_end != $end_pos)) {
       next;
     }
     # Match for structural variants
     if($sv == 1){
-      next unless ($seq_region_start >= $new_start && $seq_region_start <= $start_max_pos && $seq_region_end >= $end_min_pos && $seq_region_end <= $new_end);
-    }
-
-    # All datasets where variant was found
-    my $datasets_found = $vf->get_all_VariationSets();
-    foreach my $set (@$datasets_found) {
-      $dataset_var_found{$set->dbID()} = $set;
+      next unless ($seq_region_start >= $start_pos && $seq_region_start <= $start_max_pos && $seq_region_end >= $end_min_pos && $seq_region_end <= $end_pos);
```

Comment:
unless is to be avoided.
unless >= is equivalent to if <, just a lot harder to read.

Reply:
Thanks for the suggestions.